arch/arm64/boot/dts/aspeed-bmc-amd-marley.dts

```diff
@@ -0,0 +1,449 @@
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+
+		i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+
+		i2c@4 {
+			reg = <4>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+	};
+};
+
+&gpio0 {
+	gpio-line-names =
+		/*A0-A7*/       "","","","","","","","",
+		/*B0-B7*/       "","","","","","","","",
+		/*C0-C7*/       "","","","","","","","",
+		/*D0-D7*/       "","","","","","","","",
+		/*E0-E7*/       "","","","","","","","",
+		/*F0-F7*/       "","","","","","","","",
+		/*G0-G7*/       "","","","","","","","",
+		/*H0-H7*/       "","","","","","","","",
+		/*I0-I7*/       "","","","","","","","",
+		/*J0-J7*/       "","","","","","","","",
+		/*K0-K7*/       "","","","","","","","",
+		/*L0-L7*/       "","","","","","","","",
+		/*M0-M7*/       "","","","","","","","",
+		/*N0-N7*/       "","","","","","","","",
+		/*O0-O7*/       "","","","","","","","",
+		/*P0-P7*/       "","","","","","","","",
+		/*Q0-Q7*/       "","","","","","","","",
+		/*R0-R7*/       "","","","","","","","",
+		/*S0-S7*/       "","","","","","","","",
+		/*T0-T7*/       "","","","","","","","",
+		/*U0-U7*/       "","","","","","","","",
+		/*V0-V7*/       "","","","","","","","",
+		/*W0-W7*/       "","","","","","","","",
+		/*X0-X7*/       "","","","","","","","",
+		/*Y0-Y7*/       "","","","","","","","",
+		/*Z0-Z7*/       "","","","","","","","",
+		/*AA0-AA7*/     "","","","","","","","",
+		/*AB0-AB7*/     "ASSERT_BMC_READY","P0_MGMT_ASSERT_CLR_CMOS","P0_MGMT_MON_PROCHOT_L","","","","P0_MGMT_ASSERT_NMI_BTN_L","MGMT_SYS_MON_ATX_PWR_OK",
```

Comment:
split into 2line for readabilty point , you can refer othere device tree definitions , for better readbailty.